lib/product.rb

```diff
@@ -0,0 +1,30 @@
+class Product
+
+  attr_reader :category, :name, :unit_price
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity
+  end
+
+  def quantity
+    @quantity.to_i
+  end
+
+  def total_price
+    @unit_price * quantity
+  end
+
+  def is_hoarded?
+    if quantity >= 30
```

Comment:
I would read the interaction pattern and readme more closely. It looks like you have the hoard logic mixed with seeing if a cart is full